### PR TITLE
Update main-gpu.py

### PR DESCRIPTION
**Pull request by Kongsea**
The original one won't obtain the grads of the input. So update it.
Please check it.
ref: https://discuss.pytorch.org/t/how-to-get-cuda-variable-gradient/1386

### CODE REVIEW

**Comment by yunjey** [comment]
@Kongsea Thanks. Can you delete `requires_grad=True` in images? It is unnecessary.

**Comment by Kongsea** [comment]
@yunjey Now it seems OK. Please check it again. Thank you.

**Comment by yunjey** [comment]
@Kongsea This branch has confilcts with master branch. Could you close and reopen the pull request?

**Comment by Kongsea** [comment]
OK
